src/main/java/net/atos/entng/calendar/ical/ICalHandler.java

```diff
@@ -331,4 +319,29 @@ private void addEvent(Calendar calendar, String startDate, String endDate, Strin
         }
     }
 
+    /* Return something that can handle errors ? */
+    private String formatIcsDate (java.util.Date date) throws UnhandledEventException {
+        java.util.Date newStartDate;
+        try {
+            newStartDate = new SimpleDateFormat(DateUtils.ICAL_DATE_FORMAT).parse(date.toString());
+        } catch (ParseException e) {
+            newStartDate = formatIcsAlldayDate(date, e);
+        }
+        return new SimpleDateFormat(DateUtils.DATE_FORMAT_UTC).format(newStartDate);
+    }
+
+    private java.util.Date formatIcsAlldayDate(java.util.Date date, ParseException e1) throws UnhandledEventException {
+        java.util.Date newStartDate;
+        try {
+            newStartDate = new SimpleDateFormat(DateUtils.ICAL_ALLDAY_FORMAT).parse(date.toString());
+        } catch (ParseException e2) {
+            String message1 = String.format("[Calendar@%s::formatIcsAlldayDate] Errors have occured" +
+                            " during ical import: %s and %s",
+                    this.getClass().getSimpleName(), e1.getMessage(), e2.getMessage());
```

Comment:
Si tu passe en static this.getClass().getSimpleName() -> ICalHandler.class.getSimpleName()